sources/src/main/java/com/google/solutions/jitaccess/core/adapters/SlackAdapter.java

```diff
@@ -0,0 +1,52 @@
+package com.google.solutions.jitaccess.core.adapters;
+
+import java.net.HttpURLConnection;
+
+import com.google.common.base.Preconditions;
+import com.slack.api.Slack;
+import com.slack.api.webhook.Payload;
+import com.slack.api.webhook.WebhookResponse;
+
+public class SlackAdapter {
+  private final Options options;
+  private final Slack slack;
+
+  public SlackAdapter(Options options) {
+    Preconditions.checkNotNull(options, "options");
+    this.options = options;
+    this.slack = Slack.getInstance();
+  }
+
+  public void sendSlackMessage(
+    String content
+  ) throws SlackException {
+    Preconditions.checkNotNull(content, "content");
+
+    try {
+      WebhookResponse response = slack.send(this.options.slackHookUrl, Payload.builder().text(content).build());
+      if (!response.getCode().equals(HttpURLConnection.HTTP_OK)) {
+        throw new SlackException("Failed to send Slack notification. HTTP Code: " + response.getCode(), null);
+      }
+    } catch (Exception e) {
+      throw new SlackException("Failed to send Slack notification", e);
```

Comment:
Prefer to use a complete sentence as error message... such as _The Slack notification could not be sent_

Reply:
Adjusted!